### PR TITLE
Move database.yml to database.tmpl.yml and copy it for CI

### PR DESCRIPTION
**Pull request by bdunne**
- Allow people to maintain their own database.yml
- Copy database.yml from template if it doesn't exist

[I think I told GM that I would do this](https://github.com/ManageIQ/manageiq-schema/pull/452#issuecomment-622374578)

### CODE REVIEW

**Comment by agrare** [comment]
:+1:

**Comment by bdunne** [comment]
> I think the database.yml needs to be added to .gitignore. also should we update the readme since the instructions don't quite match?

All of `spec/dummy/` is already in the `.gitignore`.  README updated.

File: README.md [hunk 1]
```diff
@@ -24,9 +24,10 @@ The gem is available as open source under the terms of the [Apache License 2.0](
 5. Create new Pull Request
 
 ## Run Tests in dev environment
-1. Modify `spec/dummy/config/database.yml` to access local database
+1. If necessary, copy `spec/dummy/config/database.tmpl.yml` to `spec/dummy/config/database.yml` and modify it to access your local database
 2. Run: `bin/setup`
-    1. Performs `bundle update`
-    2. Generates random database region number
-    3. Creates/resets `dummy_test` database
+    1. Copies `spec/dummy/config/database.tmpl.yml` to `spec/dummy/config/database.yml` it it doesn't exist
```
Comment (chessbyte) [inline README.md:29]: typo: `it it`

Comment (bdunne) [reply README.md]: Fixed, thanks!

**Pull request merged**